test/suites/smoke/test-staking-rewards.ts

```diff
@@ -347,66 +364,102 @@ describeSuite({
-      for (;;) {
-        const blockHash = await api.rpc.chain.getBlockHash(iterOriginalRoundBlock);
-        const round = await (await api.at(blockHash)).query.parachainStaking.round();
-        if (
-          round.current.eq(originalRoundNumber) ||
-          iterOriginalRoundBlock.sub(round.length).toNumber() < 0
-        ) {
-          break;
+      // Loads information and APIs for a given round
+      const loadRoundData = async (blockNumber: BN): Promise<RoundData> => {
+        console.log(`Loading round data for block number ${blockNumber}`);
+        let firstBlockNumber: BN;
+        let blockHash = await api.rpc.chain.getBlockHash(blockNumber);
+        let apiAt = await api.at(blockHash);
+        const data = await apiAt.query.parachainStaking.round();
+        // update the block number and api with the right block
+        if (!data.first.toBn().eq(blockNumber)) {
+          firstBlockNumber = data.first.toBn();
+          blockHash = await api.rpc.chain.getBlockHash(data.first);
+          apiAt = await api.at(blockHash);
+        } else {
+          firstBlockNumber = blockNumber;
         }
 
-        // Go to previous round
-        iterOriginalRoundBlock = iterOriginalRoundBlock.sub(round.length);
-      }
-      // Go to the last block of the (original round - 1) since data is snapshotted at round start.
-      const originalRoundPriorBlock = iterOriginalRoundBlock.subn(1);
-      const originalRoundPriorBlockHash = await api.rpc.chain.getBlockHash(originalRoundPriorBlock);
-      const apiAtOriginal = await api.at(originalRoundPriorBlockHash);
+        const priorBlockHash = await api.rpc.chain.getBlockHash(firstBlockNumber.subn(1));
+        let priorApiAt = await api.at(priorBlockHash);
+
+        return {
+          data,
+          firstBlock: (await api.rpc.chain.getBlock(blockHash)).block,
+          firstBlockApi: apiAt,
+          firstBlockSpecVersion: (await apiAt.query.system.lastRuntimeUpgrade())
+            .unwrap()
+            .specVersion.toNumber(),
+          priorBlock: (await api.rpc.chain.getBlock(priorBlockHash)).block,
+          priorBlockApi: priorApiAt,
+        };
+      };
+
+      // Loads all information related to rounds involved in the reward
+      const loadPaymentRounds = async (blockNumber: BN): Promise<PaymentRounds> => {
+        const rewardRound = await loadRoundData(blockNumber);
+
+        const rewardDelay = rewardRound.firstBlockApi.consts.parachainStaking.rewardPaymentDelay;
+
+        const roundToPay = await loadRoundData(
+          (await getPreviousRound(api, rewardRound.data, rewardDelay)).first.toBn()
+        );
+        const delayedPayoutRound = await loadRoundData(
+          (await getNextRound(api, roundToPay.data)).first.toBn()
+        );
+
+        // Payment of collators have been moved 1 block later at runtime 2100
+        const delayBlockReward = rewardRound.firstBlockSpecVersion >= 2100 ? new BN(1) : new BN(1);
```

Comment:
What's the purpose of the ternary if it's the same result?

Reply:
This is a mistake, I'll replace it, but it is more informative than useful as it won't support runtime 2700 and less